### DISCUSSION TITLE
Uncaught exception 'Error' with message 'Call to a member function matching() on null'

### DISCUSSION THREAD

**Discussion by muratdemir0** [doctrine/doctrine2#10484] [Support Questions]
Hello folks, 

I try to upgrade orm from 2.4.8 to 2.7.5 however I'm getting an error on production env. If you have any idea or suggestion that would be more helpful.

My setup
PHP: 7.4.33
symfony: 2.8.52
doctrine/orm: 2.7.5
doctrine/doctrine-bundle: 1.9.1
doctrine/deprecations: 1.0.0
doctrine/annotations: 1.14.2
doctrine/reflection: 1.2.3
doctrine/event-manager: 1.2.0
doctrine/collections: 1.8.0
doctrine/cache: 1.13.0
doctrine/persistence: 1.3.8
doctrine/inflector: 1.4.4
doctrine/common: 2.13.3
doctrine/instantiator: 1.5.0
doctrine/dbal: 2.13.9

USED:

`    public function getActiveProductEstimatedDelivery()
    {
        return $this->getProductEstimatedDeliveries()
            ->matching(
                Criteria::create()
                    ->where(Criteria::expr()->eq('active', true))
            )->first();
    }
`
       
ERROR:
`Error: Uncaught exception 'Error' with message 'Call to a member function matching() on null' in /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/PersistentCollection.php:652
in Doctrine\ORM\PersistentCollection::matching called at /var/www/html/src/Modanisa/Bundle/CoreBundle/Entity/Product/ProductFilter.php (246)`

**Comment by mpdude** [comment]
Find out why the `getProductEstimatedDeliveries()` method returns a `null` value.

**Reply by muratdemir0** [reply]
the method is initialized as an array collection in the constructor method. 

`    /**
     * ProductFilter constructor.
     */
    public function __construct()
    {
        $this->productEstimatedDeliveries = new ArrayCollection();
    }`

**Reply by mpdude** [reply]
And what does the `getProductEstimatedDeliveries` method return?
